cluster-controller/src/test/java/io/strimzi/controller/cluster/model/KafkaConnectClusterTest.java

```diff
@@ -181,4 +181,53 @@ public void testGenerateDeployment()   {
         assertEquals(new Integer(1), dep.getSpec().getStrategy().getRollingUpdate().getMaxSurge().getIntVal());
         assertEquals(new Integer(0), dep.getSpec().getStrategy().getRollingUpdate().getMaxUnavailable().getIntVal());
     }
+
+    @Test
+    public void testCorruptedValues() {
+        ConfigMap cm = ResourceUtils.createEmptyKafkaConnectClusterConfigMap(namespace, cluster);
+
+        // type mismatch
+        cm.getData().put("kafka-healthcheck-delay", "1z");
+        try {
+            KafkaCluster kc = KafkaCluster.fromConfigMap(cm);
+        } catch (Throwable t) {
```

Comment:
This doesn't actually test that `fromConfigMap()` is throwing anything – if it doesn't throw the `catch` will be skipped and the test will still pass. Also you should only catch the specific exception that you want to assert. So you need to write it like this:

```
try {
    ...
    fail("Expected it to throw");
} catch (InvalidConfigMapException e) {
    // any assertions on message
}
```

The other way you can do it is to write `@Test(expect=InvalidConfigMapException.class)`-annotated methods for each of the fields. You end up with more tests that way (so it's more verbose), but each of them are more specific. 

Generally speaking it's a bad idea to assert equality on an exception message and it makes changing the exception message very painful is it will break all the tests. In this case `InvalidConfigMapException` should have a property for the field as well as the message. Then you can assert that the exception is about the expected key, without caring how the message is worded. It will also fit with the events stuff when we get to it.